Documentation: Add docs to build with llama2c (#117)
nnstreamer/nnstreamer-android: documentation/getting_started.md

./ml_inference_offloading/build/outputs/apk/release/ml_inference_offloading-release-unsigned.apk
```

#### Building the Android App with llama2

> If you want to provide LLM task(llama2) the following steps are required.

Reviewer: ```suggestion
The following steps are required to run an LLM task (i.e., llama2.c) via the Android app.
```

Author: Updated!
Thank you :)

##### Preparing the llama2 model

> We need model and tokenizer file for LLM task. You can download small model for android device using this [guide](https://github.com/karpathy/llama2.c/tree/master?tab=readme-ov-file#custom-tokenizers).

Reviewer: ```suggestion
We need a pair of the model and tokenizer files for an LLM task. You can download a small model for the Android device using this [guide](https://github.com/karpathy/llama2.c/tree/master?tab=readme-ov-file#custom-tokenizers).
```

Author: Updated!
Thank you :)

```bash
$ ls
model.bin tok4096.bin

$ mv model.bin nnstreamer-android/ml_inference_offloading/src/main/assets/models/llama2c.bin
$ mv tok4096.bin nnstreamer-android/ml_inference_offloading/src/main/assets/models/tokenizer.bin
```

> ##### Preparing the llama2.c android

Reviewer: ```suggestion
##### Building the llama2.c for Android
```

Author: Updated!
Thank you :)

To build application with llama2.c, you need to clone this [repository](https://github.com/nnsuite/llama2.c).

```bash
$ git clone https://github.com/nnsuite/llama2.c externals/llama2.c

> 

Author: I also thought about a submodule, but I was wondering if it would be a good idea to include it by default.

> 

Reviewer: I am not sure. Importing it as a git-submodule but disabling the option by default is my idea...

> 

Author: https://github.com/nnstreamer/nnstreamer-android/pull/120

Added llama2c as a git submodule.
https://github.com/nnstreamer/nnstreamer-android/blob/af85731817253510592111ec1ea973e4d65c29aa/gradle.properties#L9

By the way, `gradle.properties` has a dir.llama2c as default now

> 

Reviewer: > #120
> 
> Added llama2c as a git submodule.
> 
> https://github.com/nnstreamer/nnstreamer-android/blob/af85731817253510592111ec1ea973e4d65c29aa/gradle.properties#L9
> 
> By the way, `gradle.properties` has enabling dir.llama2c as default now

Do you mean enabling, not disabling?

> 

Author: Yes, so I commented out `dir.llama2c` in `gradle.properties`.
Could it be that this was not your intention?

> 

Reviewer: Got it. I misunderstood the previous comment.

$ ls externals/llama2.c/
...omitted...
```

> Then set llama2.c repository to the [gradle.properties](/gradle.properties).

Reviewer: ```suggestion
Then, set the directory where the llama2.c placed to the  [gradle.properties](/gradle.properties).
```

```
dir.llama2c=llama2.c
```

> Now your android application will be built with llama2.c.

Reviewer: ```suggestion
```
